src/i18n.js

```diff
@@ -0,0 +1,78 @@
+
+fs = require("fs");
+
+const extract = () => {
+
+    var output =
+        "# Gettext Message File for Plone mockup\n" +
+        'msgid ""\n' +
+        'msgstr ""\n' +
+        '"Project-Id-Version: mockup\\n"\n' +
+        '"Last-Translator: Plone i18n <plone-i18n@lists.sourceforge.net>\\n"\n' +
+        '"Language-Team: Plone i18n <plone-i18n@lists.sourceforge.net>\\n"\n' +
+        '"MIME-Version: 1.0\\n"\n' +
+        '"Content-Type: text/plain; charset=utf-8\\n"\n' +
+        '"Content-Transfer-Encoding: 8bit\\n"\n' +
+        '"Plural-Forms: nplurals=1; plural=0;\\n"\n' +
+        '"Language-Code: en\\n"\n' +
+        '"Language-Name: English\\n"\n' +
+        '"Preferred-Encodings: utf-8\\n"\n' +
+        '"Domain: widgets\\n"\n\n';
+
+    var okayFiles = ["xml", "js", "htm", "html"];
+    var found = [];
+    var checkFile = function (filepath) {
+        var split = filepath.split(".");
+        if (okayFiles.indexOf(split[split.length - 1]) === -1) {
+            return;
+        }
+        console.log("reading file: " + filepath);
+        var file = fs.readFileSync(filepath, { encoding: "utf-8" });
+        var re = /_t\((("[^"]+")|('[^']+'))(,\W{.*})?\)?\)/g;
+        var m = re.exec(file);
+        while (m) {
+            if (m) {
+                re.lastIndex = Math.max(m.index + 1, re.lastIndex + 1);
+                var val = m[1].replace(/['"]+/g, "");
+                if (found.indexOf(val) === -1) {
+                    output +=
+                        "#: " +
+                        filepath +
+                        "\n" +
+                        'msgid "' +
+                        val +
+                        '"\n' +
+                        'msgstr ""\n\n';
+                    found.push(val);
+                }
+                m = re.exec(file);
+            }
+        }
+    };
+    var checkDir = function (path) {
+        console.log("read folder: " + path);
+        var files = fs.readdirSync(path);
+        files.forEach(function (filename) {
+            if (
+                filename === "__pycache__" ||
+                filename === "build" ||
+                filename === "coverage" ||
+                filename === "less" ||
+                filename === "node_modules" ||
+                filename === "tests"
+            ) {
+                return;
+            }
+            var stats = fs.statSync(path + filename);
+            if (stats.isDirectory()) {
+                checkDir(path + filename + "/");
+            } else {
+                checkFile(path + filename);
+            }
+        });
+    };
+    checkDir("./");
+    fs.writeFileSync("../widgets.pot", output);
```

Comment:
This needs to be `widgets.pot` without `../`.
At least, on my system, the pot file ends up outside of the repository, in the directory just above it. It does not matter whether I am in `mockup` or in `mockup/src`. I guess npm changes to the directory where the `package.json` lives`.